### ISSUE TITLE
Possible to do JSX props autocomplete?

### ISSUE THREAD

**Issue by zerkalica** [steelbrain/flow-ide#38]
It's possible to do jsx props autocomplete ?

```js
// @flow
function A(props: {a: string}) {
    return <div/>
}
function B() {
    return <A/> // Ctrl+Enter does not autocomplete A props
}
```

**Comment by steelbrain** [comment]
I wish we could do that, but Flow doesn't provide that kind-of information for JSX values. We could get it from flow by a series of requests but it would get too slow and rather unnecessary.

Soon you'll have the ability to ctrl+click or option+click on a variable to goto it's declaration so I suppose you won't even need this :)

**Comment by zerkalica** [comment]
Do you mean hyperclick plugin from nuclide or something else? Hyperclick unusable with reimports.

``js
// @flow
// My.js
export class My {}
```

```js
// @flow
// a.js
import {My} from './My'

export {My}
```
```js
// @flow
b.js
import {My} from './a'
const my = new My() // jumping to My is not working
```

**Comment by steelbrain** [comment]
I didn't mean a hyperclick plugin, I meant a [declarations](https://atom.io/packages/declarations) plugin :)

and not necessarily a plugin, just support in this package

**Comment by stunaz** [comment]
Any updates on this?

**Comment by steelbrain** [comment]
Yep! We support JSX props autocomplete now, but to have the autocomplete you have to make Flow understand the signature of the components, for example

```js
type Props = {
  name: string,
}

const MyComponent = (props: Props) => (
  <div>{props.name}</div>
)
<MyComponent na/* suggestions will appear here */
```

**Comment by stunaz** [comment]
nope, it does work for me,  used your example above. not that I am using flow `0.53.1` and the language of the file is market `Babel es6 Javascript`
